server/channels/store/sqlstore/post_store.go

```diff
@@ -2350,17 +2397,45 @@ func (s *SqlPostStore) AnalyticsPostCountsByDay(options *model.AnalyticsPostCoun
 	args = append(args, end, start)
 
 	rows := model.AnalyticsRows{}
-	err := s.GetReplica().Select(
-		&rows,
-		query,
-		args...)
+	err := s.GetReplica().Select(&rows, query, args...)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to find Posts with teamId=%s", options.TeamId)
 	}
 	return rows, nil
 }
 
+func (s *SqlPostStore) countByTeam(teamID string) (int64, error) {
+	query := s.getQueryBuilder().
+		Select("COALESCE(SUM(num), 0) AS total").
+		From("posts_by_team_day")
+
+	if teamID != "" {
+		query = query.Where(sq.Eq{"teamid": teamID})
+	}
+
+	var v int64
+	err := s.GetReplica().GetBuilder(&v, query)
+	if err != nil {
+		return 0, fmt.Errorf("failed to count Posts by team: %w, teamID: %s", err, teamID)
+	}
+
+	return v, nil
+}
+
 func (s *SqlPostStore) AnalyticsPostCount(options *model.PostCountOptions) (int64, error) {
+	if s.DriverName() == model.DatabaseDriverPostgres {
+		// Only call this method if no other options are passed.
+		// This is specialized for system console stats.
+		if !options.UsersPostsOnly &&
+			!options.MustHaveFile &&
+			!options.MustHaveHashtag &&
+			!options.ExcludeDeleted &&
+			!options.ExcludeSystemPosts &&
+			options.SinceUpdateAt == 0 &&
+			options.UntilUpdateAt == 0 {
```

Comment:
So, if we were to add a new option to `model.PostCountOptions`, we'd likely have to update this check to avoid a bug.

More in general, what would be our strategy if some day we wanted to surface this sort of filtering capabilities to the admins while retaining the performance boost?